app/models/user.rb

```diff
@@ -33,6 +33,6 @@ def name
   end
 
   def lead?
-    memberships.where(role: 1).any?
+    memberships.where(role: 'lead').any?
```

Comment:
I think this would avoid loading all memberships into memory, and just check with SQL... I think.

```suggestion
    memberships.where(role: 'lead').exists
```

Reply:
Changed this but just read in Rails 5.1 and up, the performance of any? and exists? has been fixed, so they are now equally performant.